src/board/generic_esp32/generic-esp32-board.c

```diff
@@ -480,15 +541,38 @@ void SX126xWriteRegister( uint16_t address, uint8_t value )
  */
 uint8_t SX126xReadRegister( uint16_t address )
 {
-    return 0;    
+    uint8_t data;
+    SX126xReadRegisters(address, &data, 1);
+    return data;
 }
 
 void SX126xReadRegisters( uint16_t address, uint8_t *buffer, uint16_t size )
 {
-    
+    spi_transaction_t trans = {};
+    trans.cmd = RADIO_READ_REGISTER;
+    trans.addr = address;
+    trans.rx_buffer = buffer;
+    trans.length = size * 8;
+
+    SX126xCheckDeviceReady();
+
+    ESP_ERROR_CHECK(spi_device_transmit(lora_spi.handle, &trans));
+
+    SX126xWaitOnBusy();
 }
 void SX126xWriteRegisters( uint16_t address, uint8_t *buffer, uint16_t size )
 {
+    spi_transaction_t trans = {};
+    trans.cmd = RADIO_WRITE_REGISTER;
+    trans.addr = address;
+    trans.tx_buffer = buffer;
+    trans.length = size * 8;
+
+    SX126xCheckDeviceReady();
```

Comment:
what if this never indicates ready? if we are ok with blocking at this point we should note the behavior in the README.md of this component to aid implementers in debugging

Reply:
added this, and additional detail to the README